tests/test_main.py

```diff
@@ -0,0 +1,124 @@
+"""Run from the `pyta` root directory to use the local `python_ta` rather than
+installed `python_ta` package.
+"""
+
+import subprocess
+import sys
+from os import environ
+
+import python_ta
+
+
+def test_check_no_errors_zero() -> None:
+    """Test that python_ta exits with status code 0 when it does not detect errors."""
+    output = subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "python_ta",
+            "--config",
+            "tests/test.pylintrc",
+            "tests/fixtures/no_errors.py",
+        ]
+    )
+
+    assert output.returncode == 0
+
+
+def test_check_errors_nonzero() -> None:
+    """Test that python_ta exits with non-zero status code when it detects errors."""
+    output = subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "python_ta",
+            "--config",
+            "tests/test.pylintrc",
+            "examples/nodes/name.py",
+        ]
+    )
+
+    assert output.returncode != 0
+
+
+def test_check_exit_zero() -> None:
+    """Test that python_ta --exit-zero always exits with status code 0,
+    even when given a file with errors.
+    """
+    output = subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "python_ta",
+            "--exit-zero",
+            "--config",
+            "tests/test.pylintrc",
+            "examples/nodes/name.py",
+        ],
+        env={**environ, "PYTHONIOENCODING": "utf-8"},
+    )
+
+    assert output.returncode == 0
+
+
+def test_check_version() -> None:
+    """Test that python_ta --version outputs python_ta.__version__ to stdout."""
+    stdout = subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "python_ta",
+            "--config",
+            "tests/test.pylintrc",
+            "--version",
+        ],
+        capture_output=True,
+        text=True,
+    ).stdout
+
+    assert stdout.rstrip("\n") == python_ta.__version__
+
+
+def test_config_generation() -> None:
+    """Test that python_ta --generate-config prints the default config to stdout."""
+    stdout_config = subprocess.run(
+        [
+            sys.executable,
+            "-m",
+            "python_ta",
+            "--generate-config",
+            "--config",
+            "tests/test.pylintrc",
+        ],
+        capture_output=True,
+        text=True,
+    ).stdout.rstrip("\n")
+
+    actual_config = subprocess.run(
```

Comment:
You don't need subprocess here. Instead, you can just read the contents of that file (i.e., use the file `read` method). To avoid the problem of relative paths, talk to @Raine-Yang-UofT about what I suggested to him and use the same approach here.

Reply:
I was slightly concerned about the relative paths but I am actually somewhat more concerned about the fact that the paths are hard-coded. 

Is there a way to set up a global constant defining, for instance, the path to the default config file?

Reply:
Though to be clear, such a constant should only include the relative path from the source code root to the config file. So you'll still need to avoid the problem of relative paths by following what I mentioned above.

Reply:
I wrote @Raine-Yang-UofT but he has not yet gotten back to me. Did you put that information in a public comment on his PR as well?

Reply:
Did you mean I should introduce a constant and store it under `python_ta/config` or that the constant should store the path `python_ta/config`. 

I initially interpreted your comment as the former, but I have found that this interpretation doesn't seem practical given what is currently stored in that location. Also, the `.pylintrc` config file is right in that exact directory, so storing such a constant there, right next to the file it points to, seems a bit redundant.

Reply:
> Though to be clear, such a constant should only include the relative path from the source code root to the config file. So you'll still need to avoid the problem of relative paths by following what I mentioned above.

I think this is the only thing I am now missing. I am not sure what the approach you would recommend is for avoiding the relative path issue. What I would imagine is it involves some way of finding the source code root from anywhere in the project directory structure, such that a relative path from the source root to anywhere is still useful, since we have now found the path to the source root. I hope that makes sense...

Reply:
> > Though to be clear, such a constant should only include the relative path from the source code root to the config file. So you'll still need to avoid the problem of relative paths by following what I mentioned above.
> 
> I think this is the only thing I am now missing. I am not sure what the approach you would recommend is for avoiding the relative path issue. What I would imagine is it involves some way of finding the source code root from anywhere in the project directory structure, such that a relative path from the source root to anywhere is still useful, since we have now found the path to the source root. I hope that makes sense...

Ok, he got back to me. I believe I understand your reasoning for asking for that but I think in some cases a more generalized approach may be more beneficial. I will submit what I mean shortly in a new commit...